src/runtime_src/core/common/xclbin_parser.cpp

```diff
@@ -393,13 +393,14 @@ get_softkernels(const axlf* top)
       auto begin = reinterpret_cast<const char*>(top) + pSection->m_sectionOffset;
       auto soft = reinterpret_cast<const soft_kernel*>(begin);
 
-      softkernel_object sko;
+      sks.emplace_back(softkernel_object{});
+      auto& sko = sks.back();
```

Comment:
This is not exception safe.  Please don't modify sks before the sko object has been created *and* populated successfully.
Move the emplace to the end as I suggested before.